### ISSUE TITLE
JSDoc is not considered when variable type is displayed.

### ISSUE THREAD

**Issue by dotnetCarpenter** [Microsoft/vscode#54679]
Issue Type: <b>Bug</b>

I have a function (`composePromise`) that I import with JSDoc documentation specifying the return type, but vscode ignores the return type and sets it to `any`. The type MUST BE `function`. This affects the color coding used (Dark+).

The code that is imported:
```js
/**
 * Compose functions that can return a Promise.
 * If a Promise is returned then the next function will not be invoked until
 * the Promise is resolved.
 * @param {function[]} fs functions that can return a Promise
 * @returns {function} A function that, when called, will start the chain of function invocations. Can be started with a Promise.
 */
export default function compose (...fs) { return initialValue => fs.reduceRight(
  (x, f) => Promise.resolve(x).then(f),
  initialValue
) }
```

The calling code:
```js
import composePromise from '../utils/compose.promise.mjs'

// createFlow is a function
const createflow = composePromise(
  annotations => annotations.filter(isNappAnnotation),
  annotations => annotations.map(compose(
    createNappAnnotationWithInstance,
    identity
  )),
  partial(getAnnotations, instance),
  removeFlow
)
```
![wrong-return-type](https://user-images.githubusercontent.com/43763/42958687-97395574-8b86-11e8-9f5d-cdd2c5fa9b8f.png)

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:40:20.190Z)
OS version: Linux x64 4.15.0-24-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Celeron(R) CPU  N2840  @ 2.16GHz (2 x 2582)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: disabled_off<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: unavailable_software<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 1|
|Memory (System)|3.74GB (0.17GB free)|
|Process Argv|/usr/share/code/code frontend/|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
vetur|oct|0.12.5

</details>

Changing name of a default export is covered here: http://2ality.com/2014/09/es6-modules-final.html#the-default-export-is-just-another-named-export

**Comment by mjbvz** [comment]
Tested in VS Code insiders. This seems to detect that you are returning a function instead of any. But `@return {function}` explicitly says that you return a generic function. If you want to set the type of the function, write something like:

```
@returns {(x: number) => string}
```

or

```
@returns {function(number): string}
```

**Comment by dotnetCarpenter** [comment]
@mjbvz So in this case it would be `@returns {function(...function): Promise}` or `{(...fs:function) => Promise}`?

I have to take your word that it's fixed in the upcoming vscode, because in `1.25.1` it doesn't work.
